### PR TITLE
deploy: update csi-snapshotter sidecar to v5.0.1

### PR DESCRIPTION
**Pull request by humblec**
This release of snapshotter has a breaking change as mentioned
in the release note:

Refer# https://github.com/kubernetes-csi/external-snapshotter/releases/tag/v5.0.0

RBAC rules are also updated with this commit.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

### CODE REVIEW

**Comment by nixpanic** [comment]
snapshot component installation seems to fail?

```
error: the namespace from the provided object "kube-system" does not match the namespace "default". You must pass '--namespace=kube-system' to perform this operation.
```

**Comment by humblec** [comment]
> 

Yeah, the deployment is not going through.. chceking this

File: scripts/install-snapshot.sh [hunk 1]
```diff
@@ -65,8 +65,8 @@ function create_or_delete_resource() {
     mkdir -p "${TEMP_DIR}"
     curl -o "${temp_rbac}" "${SNAPSHOT_RBAC}"
     curl -o "${temp_snap_controller}" "${SNAPSHOT_CONTROLLER}"
-    sed -i "s/namespace: default/namespace: ${namespace}/g" "${temp_rbac}"
-    sed -i "s/namespace: default/namespace: ${namespace}/g" "${snapshotter_psp}"
+    sed -i "s/namespace: kube-system/namespace: ${namespace}/g" "${temp_rbac}"
+    sed -i "s/namespace: kube-system/namespace: ${namespace}/g" "${snapshotter_psp}"
     sed -i "s/canary/${SNAPSHOT_VERSION}/g" "${temp_snap_controller}"
```
Comment (Madhu-1) [inline scripts/install-snapshot.sh:71]: namespace change is required for controller yaml also

**Review by nixpanic — DISMISSED** [review]

**Comment by nixpanic** [comment]
Added to the 3.6 milestone, it is important for users to know about this change and update the RBAC rlues.

**Review by Madhu-1 — CHANGES_REQUESTED** [review]

File: scripts/install-snapshot.sh (hunk 1 above)
Comment (humblec) [reply scripts/install-snapshot.sh:71]: @Madhu-1 added there as well.. ptal.

**Review by Madhu-1 — APPROVED** [review]

**Review by yati1998 — APPROVED** [review]

**Comment by humblec** [comment]
@Mergifyio refresh

**Comment by humblec** [comment]
@Mergifyio rebase

**Comment by humblec** [comment]
@Mergifyio refresh

**Comment by humblec** [comment]
@Mergifyio refresh

**Comment by humblec** [comment]
@Mergifyio rebase

**Comment by humblec** [comment]
@Mergifyio refresh

**Comment by humblec** [comment]
@Mergifyio rebase

**Comment by humblec** [comment]
@Mergifyio refresh

**Comment by nixpanic** [comment]
@mergifyio rebase

**Pull request merged**